### ISSUE TITLE
`grant_expires_in` is not honoured

### ISSUE THREAD

**Issue by lwm** [OpenIDC/pyoidc#328]
The `SessionDB` takes a `grant_expires_in` kwarg which doesn't seem to be working.

I had to patch the `oic.oic.provider.token_endpoint` with the following:

``` python
    def token_endpoint(self, request="", **kwargs):
        data = dict(parse_qsl(request))

        try:
            event = self.sdb[data['code']]['authn_event']
        except KeyError:
            return Response(status="400 Bad Request")

        if int(time()) > event.valid_until:
            return Response(status="498 Invalid Token")

        return super().token_endpoint(request, **kwargs)
```

I don't see any tests covering this, so I think this might be an oversight?

**Comment by lwm** [comment]
Probably `int(time()) > event.valid_until:` is not the right check ...

It should probably be `int(time()) > event.authn_time + self.sdb.token_factory['code'].lifetime`

**Comment by rohe** [comment]
Sorry, I've been otherwise occupied for a couple of days.

Time is extremely confusing. 
And you're right in that lifetime should have been added.

**Comment by tpazderka** [comment]
This has been fixed in #429
